pkg/sync/closer.go

```diff
@@ -78,9 +78,9 @@ func (c *Closer) IsClosed() bool {
 	return c.isClosed.IsSet()
 }
 
-// OnClose registers the passed callback to be calle when the Closer is closed.
-// If the Closer is already closed, does nothing. Returns whether the Closer was
-// not yet closed.
+// OnClose registers the passed callback to be called when the Closer is closed.
+// If the Closer is already closed, it does nothing and returns false.
+// If the Closer is not yet closed and call back was registered, returns true.
```

Comment:
```suggestion
// If the Closer is not yet closed, returns true.
```

It always returns true if the Closer is not closed, right?

(Besides that, note that callback should be written as one word.)

Reply:
Yes, it returns true if the closer is not closed.
Also, changed `call back` -> `callback`.

Reply:
OK, so if it always returns true independent of the callback, why do you mention the callback at all? (See my change suggestion above.)

Reply:
I tried to make it explicit that
1. Callback will be registered only when the channel is not yet closed. It will be ignored otherwise.

Reply:
OK, but then you should write the comment like that. Currently this is not what the comment says.